arch/arm/src/common/arm_createstack.c

```diff
@@ -237,19 +246,27 @@ int up_create_stack(FAR struct tcb_s *tcb, size_t stack_size, uint8_t ttype)
  ****************************************************************************/
 
 #ifdef CONFIG_STACK_COLORATION
-void arm_stack_color(FAR void *stackbase, size_t nbytes)
+void arm_stack_color(FAR void *stackbase, size_t size)
 {
-  /* Take extra care that we do not write outsize the stack boundaries */
+  uintptr_t start;
+  uintptr_t end;
+  FAR uint32_t *ptr;
+
+  /* Take extra care that we do not write outside the stack boundaries */
+
+  start = INT32_ALIGN_UP((uintptr_t)stackbase);
+  end   = INT32_ALIGN_DOWN((uintptr_t)stackbase + size);
+
+  /* Get the adjusted size based on the top and bottom of the stack */
 
-  uint32_t *stkptr = (uint32_t *)(((uintptr_t)stackbase + 3) & ~3);
-  uintptr_t stkend = (((uintptr_t)stackbase + nbytes) & ~3);
-  size_t    nwords = (stkend - (uintptr_t)stackbase) >> 2;
+  size = (end - start) >> 2;
```

Comment:
Re-using "size" here looks messy to me; before this line it is size in bytes, and afterwards size in 32-bit words. Maybe keep "nwords" to have it less confusing?

Reply:
Also, isn't all this sort of unnecessary? You have already aligned the stack pointer and stack size in up_createstack(), so the size in here should be 32-bit aligned. Would it be less code to e.g. simply debugassert the start pointer and the stack size alignments?

Reply:
You're right, I'll change the naming.
And yes, all the calculations are unneccesary since createstack and usestack already align the stacks properly. I wasn't sure about removing it all together... in stack_color and do_stackcheck.
I'd be fine with a debugassert as well. @davids5 what do you think?

Reply:
I think if the TCB is corrupted. I would either PANIC() (possible too draconian) or indicate 0 stack left to bubble this up. So I think I would do a Kconfig option for the PANIC and always indicate 0 stack left.